src/request.ts

```diff
@@ -203,3 +203,17 @@ filterRegistry.register(
 		};
 	}
 );
+
+/**
+ * Add a filter that automatically parses incoming Buffer responses in Node.
+ */
+filterRegistry.register(
+	function (response: Response<any>, url: string, options: RequestOptions) {
+		return Buffer.isBuffer(response.data) !== undefined && options.responseType === 'json';
```

Comment:
Also, I just looked at the code, the technically, `options` is an optional argument, so we should change the `options: RequestOptions` to `options?: RequestOptions` and safely access `options` as well.